### PR TITLE
Manual R 4.3 migration

### PR DESCRIPTION
**Pull request by mfansler**
Bot was showing unsolvability issues, so starting a manual migration to see what's happen in more detail.

## Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

**Comment by mfansler** [comment]
Issue with aarchs is that `r-base` gets installed in the `build` environment and requires libcxx=12, but compilers are pinned at 10, so it's unsolvable. Might just need to skip it? The aarchs with cxx=12 work fine.

**Comment by jakirkham** [comment]
What about just overriding the compiler versions to align?

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -151,28 +151,29 @@ outputs:
         - lib/R/lib
     requirements:
       build:
-        - {{ compiler('m2w64_c') }}          # [win]
-        - {{ compiler('m2w64_cxx') }}        # [win]
+        - libcxx                       # [aarch64 or ppc64le]
```
Comment (isuruf) [inline recipe/meta.yaml:154]: Why do you need `libcxx` here in the first place?

Comment (mfansler) [reply recipe/meta.yaml:154]: Was following what was done in [the recommended example](https://github.com/conda-forge/pp-sketchlib-feedstock/pull/40/commits/c17c59dde8a7501697a62b191e444c7267fcdf70). It seemed to imply that the `libcxx` was necessary to enforce the correct version, but maybe I misinterpreted?

@jakirkham am I doing this correctly? Doesn't actually appear to have changed the compiler version, leading to `gxx*=10` with `libcxx=12`.

Comment (jakirkham) [reply recipe/meta.yaml:154]: Yeah it's a workaround for a bug encountered with the newer Clang version

That said, think we want this on macOS

```suggestion
        - libcxx                       # [osx]      
```

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+###############################################
+## This is a temporary workaround to support 
+## R 4.3 migration. It should be removed when
+## libcxx=10 is dropped.
+c_compiler_version:    # [aarch64 or ppc64le] 
+  - 12                 # [aarch64 or ppc64le]
+cxx_compiler_version:  # [aarch64 or ppc64le] 
+  - 12                 # [aarch64 or ppc64le]
+libcxx:                # [aarch64 or ppc64le] 
+  - 12                 # [aarch64 or ppc64le]
```
[the remark this answers is not in the document]
Comment (mfansler) [reply recipe/conda_build_config.yaml:10]: However, it's not a Clang/OSX issue - it's the **linux-aarch64** and **linux-ppc64le** builds with CUDA + GCC 10 that are the issue.

**Example Failure**
```bash
RuntimeError: Solver could not find solution.Mamba failed to solve:
 - r-matrix
 - r-magrittr
 - r-knitr
 - nvcc_linux-aarch64 11.2.*
 - r-base 4.3.*
 - r-base
 - ninja
 - make
 - git
 - gcc_linux-aarch64 10.*
 - r-data.table
 - gxx_linux-aarch64 10.*
 - cmake
 - r-jsonlite
 - cross-r-base 4.3.*

with channels:

The reported errors are:
- Encountered problems while solving:
-   - package gcc_linux-aarch64-10.3.0-hd47fc70_0 requires gcc_impl_linux-64 10.3.0.*, but none of the providers can be installed
```

[the remark this answers is not in the document]
Comment (jakirkham) [reply recipe/conda_build_config.yaml:10]: Ok then I'm confused on the usage of `libcxx` here as well. Would suggest dropping that from here and `meta.yaml`

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:154]: Scratch that sounds like the issue was with GCC (misunderstood that), in which case this shouldn't be here after all

```suggestion
```

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

**Comment by mfansler** [comment]
I'm reverting my workarounds in light of [Isuru's PR](https://github.com/conda-forge/r-base-feedstock/pull/268) for a GCC 10 build of R 4.3. 🤞

**Comment by mfansler** [comment]
@isuruf fyi, looks like `cross-r-base` is the failpoint. It doesn't seem to have rebuilt in the GCC 10 PR.

**cross-r-base failure (from linux-anvil-cuda:11.2)**
<details>

```bash
$ mamba create -n foo cross-r-base=4.3 gcc_linux-aarch64=10

Looking for: ['cross-r-base=4.3', 'gcc_linux-aarch64=10']

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache
Could not solve for environment specs
The following packages are incompatible
├─ cross-r-base 4.3**  is installable and it requires
│  └─ r-base [4.3.0 h0fc540b_0|4.3.0 hfabd6f2_1|...|4.3.1 hfabd6f2_1], which requires
│     └─ gcc_impl_linux-64 >=12 , which can be installed;
└─ gcc_linux-aarch64 10**  is uninstallable because there are no viable options
   ├─ gcc_linux-aarch64 10.3.0 would require
   │  └─ gcc_impl_linux-64 10.3.0.* , which conflicts with any installable versions previously reported;
   └─ gcc_linux-aarch64 10.4.0 would require
      └─ gcc_impl_linux-64 10.4.0.* , which conflicts with any installable versions previously reported.
```

</details>

**r-base success (from linux-anvil-cuda:11.2)**

<details>

```bash
$ mamba create -dn foo r-base=4.3 gcc_linux-aarch64=10

Looking for: ['r-base=4.3', 'gcc_linux-aarch64=10']

conda-forge/linux-64                                        Using cache
conda-forge/noarch                                          Using cache
Transaction

  Prefix: /opt/conda/envs/foo

  Updating specs:

   - r-base=4.3
   - gcc_linux-aarch64=10


  Package                                           Version  Build                Channel                    Size
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────
  Install:
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  + _libgcc_mutex                                       0.1  conda_forge          conda-forge/linux-64     Cached
  + _openmp_mutex                                       4.5  2_gnu                conda-forge/linux-64     Cached
  + _r-mutex                                          1.0.1  anacondar_1          conda-forge/noarch          4kB
  + _sysroot_linux-aarch64_curr_repodata_hack             4  h57d6b7b_13          conda-forge/noarch         21kB
  + binutils_impl_linux-64                             2.40  hf600244_0           conda-forge/linux-64     Cached
  + binutils_impl_linux-aarch64                        2.39  h57c828d_1           conda-forge/linux-64        5MB
  + binutils_linux-aarch64                             2.39  h9a81858_13          conda-forge/linux-64       28kB
  + bwidget                                          1.9.14  ha770c72_1           conda-forge/linux-64      122kB
  + bzip2                                             1.0.8  h7f98852_4           conda-forge/linux-64     Cached
  + c-ares                                           1.19.1  hd590300_0           conda-forge/linux-64     Cached
  + ca-certificates                                2023.5.7  hbcca054_0           conda-forge/linux-64     Cached
  + cairo                                            1.16.0  hbbf8b49_1016        conda-forge/linux-64        1MB
  + curl                                              8.2.0  hca28451_0           conda-forge/linux-64       91kB
  + expat                                             2.5.0  hcb278e6_1           conda-forge/linux-64      137kB
  + font-ttf-dejavu-sans-mono                          2.37  hab24e00_0           conda-forge/noarch        397kB
  + font-ttf-inconsolata                              3.000  h77eed37_0           conda-forge/noarch         97kB
  + font-ttf-source-code-pro                          2.038  h77eed37_0           conda-forge/noarch        701kB
  + font-ttf-ubuntu                                    0.83  hab24e00_0           conda-forge/noarch          2MB
  + fontconfig                                       2.14.2  h14ed4e7_0           conda-forge/linux-64      272kB
  + fonts-conda-ecosystem                                 1  0                    conda-forge/noarch          4kB
  + fonts-conda-forge                                     1  0                    conda-forge/noarch          4kB
  + freetype                                         2.12.1  hca18f0e_1           conda-forge/linux-64     Cached
  + fribidi                                          1.0.10  h36c2ea0_0           conda-forge/linux-64      114kB
  + gcc_impl_linux-64                                10.4.0  h5231bdf_19          conda-forge/linux-64       49MB
  + gcc_impl_linux-aarch64                           10.4.0  hed4fad0_19          conda-forge/linux-64      415MB
  + gcc_linux-aarch64                                10.4.0  h227fc07_13          conda-forge/linux-64       30kB
  + gettext                                          0.21.1  h27087fc_0           conda-forge/linux-64     Cached
  + gfortran_impl_linux-64                           10.4.0  h7d168d2_19          conda-forge/linux-64       15MB
  + graphite2                                        1.3.13  h58526e2_1001        conda-forge/linux-64      105kB
  + gsl                                                 2.7  he838d99_0           conda-forge/linux-64        3MB
  + gxx_impl_linux-64                                10.4.0  h5231bdf_19          conda-forge/linux-64       12MB
  + harfbuzz                                          7.3.0  hdb3a94d_0           conda-forge/linux-64        1MB
  + icu                                                72.1  hcb278e6_0           conda-forge/linux-64     Cached
  + kernel-headers_linux-64                          2.6.32  he073ed8_16          conda-forge/noarch       Cached
  + kernel-headers_linux-aarch64                     4.18.0  h5b4a56d_13          conda-forge/noarch          1MB
  + keyutils                                          1.6.1  h166bdaf_0           conda-forge/linux-64     Cached
  + krb5                                             1.21.1  h659d440_0           conda-forge/linux-64     Cached
  + ld_impl_linux-64                                   2.40  h41732ed_0           conda-forge/linux-64     Cached
  + ld_impl_linux-aarch64                              2.39  hc438fd5_1           conda-forge/linux-64      734kB
  + lerc                                              4.0.0  h27087fc_0           conda-forge/linux-64     Cached
  + libblas                                           3.9.0  17_linux64_openblas  conda-forge/linux-64       14kB
  + libcblas                                          3.9.0  17_linux64_openblas  conda-forge/linux-64       14kB
  + libcurl                                           8.2.0  hca28451_0           conda-forge/linux-64      372kB
  + libdeflate                                         1.18  h0b41bf4_0           conda-forge/linux-64     Cached
  + libedit                                    3.1.20191231  he28a2e2_2           conda-forge/linux-64     Cached
  + libev                                              4.33  h516909a_1           conda-forge/linux-64     Cached
  + libexpat                                          2.5.0  hcb278e6_1           conda-forge/linux-64     Cached
  + libffi                                            3.4.2  h7f98852_5           conda-forge/linux-64     Cached
  + libgcc-devel_linux-64                            10.4.0  hd38fd1e_19          conda-forge/linux-64        4MB
  + libgcc-devel_linux-aarch64                       10.4.0  h5d73fa8_19          conda-forge/linux-64      260kB
  + libgcc-ng                                        13.1.0  he5830b7_0           conda-forge/linux-64     Cached
  + libgfortran-ng                                   13.1.0  h69a702a_0           conda-forge/linux-64     Cached
  + libgfortran5                                     13.1.0  h15d22d2_0           conda-forge/linux-64     Cached
  + libglib                                          2.76.4  hebfc3b9_0           conda-forge/linux-64        3MB
  + libgomp                                          13.1.0  he5830b7_0           conda-forge/linux-64     Cached
  + libiconv                                           1.17  h166bdaf_0           conda-forge/linux-64     Cached
  + libjpeg-turbo                                   2.1.5.1  h0b41bf4_0           conda-forge/linux-64     Cached
  + liblapack                                         3.9.0  17_linux64_openblas  conda-forge/linux-64       14kB
  + libnghttp2                                       1.52.0  h61bc06f_0           conda-forge/linux-64     Cached
  + libopenblas                                      0.3.23  pthreads_h80387f5_0  conda-forge/linux-64        5MB
  + libpng                                           1.6.39  h753d276_0           conda-forge/linux-64     Cached
  + libsanitizer                                     10.4.0  h5246dfb_19          conda-forge/linux-64        6MB
  + libssh2                                          1.11.0  h0841786_0           conda-forge/linux-64     Cached
  + libstdcxx-devel_linux-64                         10.4.0  hd38fd1e_19          conda-forge/linux-64       10MB
  + libstdcxx-ng                                     13.1.0  hfd8a6a1_0           conda-forge/linux-64     Cached
  + libtiff                                           4.5.1  h8b53f26_0           conda-forge/linux-64     Cached
  + libuuid                                          2.38.1  h0b41bf4_0           conda-forge/linux-64     Cached
  + libwebp-base                                      1.3.1  hd590300_0           conda-forge/linux-64     Cached
  + libxcb                                             1.15  h0b41bf4_0           conda-forge/linux-64     Cached
  + libxml2                                          2.11.4  h0d562d8_0           conda-forge/linux-64     Cached
  + libzlib                                          1.2.13  hd590300_5           conda-forge/linux-64     Cached
  + make                                                4.3  hd18ef5c_1           conda-forge/linux-64      519kB
  + ncurses                                             6.4  hcb278e6_0           conda-forge/linux-64     Cached
  + openssl                                           3.1.1  hd590300_1           conda-forge/linux-64     Cached
  + pango                                           1.50.14  heaa33ce_1           conda-forge/linux-64      443kB
  + pcre2                                             10.40  hc3806b6_0           conda-forge/linux-64     Cached
  + pixman                                           0.40.0  h36c2ea0_0           conda-forge/linux-64      643kB
  + pthread-stubs                                       0.4  h36c2ea0_1001        conda-forge/linux-64        6kB
  + r-base                                            4.3.1  h29c4799_2           conda-forge/linux-64       26MB
  + readline                                            8.2  h8228510_1           conda-forge/linux-64     Cached
  + sed                                                 4.8  he412f7d_0           conda-forge/linux-64      271kB
  + sysroot_linux-64                                   2.12  he073ed8_16          conda-forge/noarch       Cached
  + sysroot_linux-aarch64                              2.17  h5b4a56d_13          conda-forge/noarch         16MB
  + tk                                               8.6.12  h27826a3_0           conda-forge/linux-64     Cached
  + tktable                                            2.10  hb7b940f_3           conda-forge/linux-64       92kB
  + xorg-kbproto                                      1.0.7  h7f98852_1002        conda-forge/linux-64       27kB
  + xorg-libice                                      1.0.10  h7f98852_0           conda-forge/linux-64       59kB
  + xorg-libsm                                        1.2.3  hd9c2040_1000        conda-forge/linux-64       26kB
  + xorg-libx11                                       1.8.6  h8ee46fc_0           conda-forge/linux-64      829kB
  + xorg-libxau                                      1.0.11  hd590300_0           conda-forge/linux-64     Cached
  + xorg-libxdmcp                                     1.1.3  h7f98852_0           conda-forge/linux-64     Cached
  + xorg-libxext                                      1.3.4  h0b41bf4_2           conda-forge/linux-64       50kB
  + xorg-libxrender                                  0.9.11  hd590300_0           conda-forge/linux-64       38kB
  + xorg-libxt                                        1.3.0  hd590300_0           conda-forge/linux-64      380kB
  + xorg-renderproto                                 0.11.1  h7f98852_1002        conda-forge/linux-64       10kB
  + xorg-xextproto                                    7.3.0  h0b41bf4_1003        conda-forge/linux-64       30kB
  + xorg-xproto                                      7.0.31  h7f98852_1007        conda-forge/linux-64       75kB
  + xz                                                5.2.6  h166bdaf_0           conda-forge/linux-64     Cached
  + zlib                                             1.2.13  hd590300_5           conda-forge/linux-64       93kB
  + zstd                                              1.5.2  hfc55251_7           conda-forge/linux-64     Cached

  Summary:

  Install: 100 packages

  Total download: 582MB

───────────────────────────────────────────────────────────────────────────────────────────────────────────────────


Dry run. Exiting.

DryRunExit: Dry run. Exiting.
```

</details>

**Comment by mfansler** [comment]
@conda-forge-admin please rerender

**Comment by mfansler** [comment]
@dpryan79 thanks for the heads up! Rerunning...

**Comment by dpryan79** [comment]
Fingers crossed, there are a bunch of recipes waiting for this on bioconda too :)

**Comment by dpryan79** [comment]
Not sure we can do anything about the CI running out of disk space.

**Comment by isuruf** [comment]
@conda-forge-admin rerender

**Comment by mfansler** [comment]
```python
conda.CondaMultiError: [Errno 28] No space left on device
```

**Comment by dpryan79** [comment]
Great, someone want to hit merge?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -151,28 +151,28 @@ outputs:
...
     requirements:
       build:
-        - {{ compiler('m2w64_c') }}          # [win]
-        - {{ compiler('m2w64_cxx') }}        # [win]
+        - {{ compiler('m2w64_c') }}    # [win]
+        - {{ compiler('m2w64_cxx') }}  # [win]
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
-        - llvm-openmp            # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml:159]: Why was this dropped?

File: recipe/meta.yaml [hunk 4]
```diff
@@ -183,7 +183,6 @@ outputs:
       run:
         - {{ pin_subpackage('libxgboost', exact=True) }}
         - {{ pin_subpackage('_r-xgboost-mutex', exact=True) }}
-        - llvm-openmp  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml:186]: Why was this dropped? Also should `libgomp` be here too?

Comment (mfansler) [reply recipe/meta.yaml:186]: `host` should suffice. `libgomp` is added to `host`.

File: recipe/meta.yaml (hunk 3 above)
Comment (mfansler) [reply recipe/meta.yaml:159]: Note that the `cuda` is not dropped - simply had the selector whitespace adjusted.

**Comment by dpryan79** [comment]
Any other comments regarding this? Otherwise perhaps it can be merged.

File: recipe/meta.yaml (hunk 3 above)
Comment (jakirkham) [reply recipe/meta.yaml:159]: Am not sure about this. For example I see this line in `openblas`:

https://github.com/conda-forge/openblas-feedstock/blob/7dbdf67050d84af131c930b26daa387e7183c7a2/recipe/meta.yaml#L46

This is also the case with `pytorch`:

https://github.com/conda-forge/pytorch-cpu-feedstock/blob/1a3257e9f1aa9cb3ac8d7347c22493024265f821/recipe/meta.yaml#L54-L55

Also a few more feedstocks:

* https://github.com/conda-forge/crest-feedstock/blob/b406e0f50edeb7d67dd0915f63d45159e8ab9c56/recipe/meta.yaml#L30-L31
* https://github.com/conda-forge/fpm-feedstock/blob/ac335692104ce7ca490093144e1da2d1922ce24c/recipe/meta.yaml#L26-L27
* https://github.com/conda-forge/pinocchio-feedstock/blob/415cc6214c2544d3f69a205148b5f3a27df23bcc/recipe/meta.yaml#L43
* https://github.com/conda-forge/pychrono-feedstock/blob/73013dff49b4c89ebf6a891995334decb8b994ce/recipe/meta.yaml#L26-L27

Could you please recheck?

Also let's leave this thread unresolved until we reach consensus. Thanks! 🙏

**Comment by jakirkham** [comment]
I've unresolved the thread above and commented in it. Let's follow up there

File: recipe/meta.yaml (hunk 4 above)
Comment (jakirkham) [reply recipe/meta.yaml:186]: Am seeing lines like this in [the CI log]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=750342&view=logs&j=cacb33e2-4f35-58d8-c6f0-bc569fa20097&t=5d96f9ff-2d0d-52ea-cc90-ffadc2f5efed&l=839 ):

```
WARNING (libxgboost,bin/xgboost): Needed DSO lib/libgomp.so.1 found in ['_openmp_mutex']
WARNING (libxgboost,bin/xgboost): .. but ['_openmp_mutex'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```

Are we sure these are not needed in `run`?

Comment (mfansler) [reply recipe/meta.yaml:186]: Apologies if I was too hasty at marking as resolved! 

Note that the `r-xgboost` section, where this PR's proposed changes are located, is clean in [the CI logs](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=750342&view=logs&j=cacb33e2-4f35-58d8-c6f0-bc569fa20097&t=5d96f9ff-2d0d-52ea-cc90-ffadc2f5efed&l=2854):

```bash
   INFO: sysroot: '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/' files: '['/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/zone.tab', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/tzdata.zi', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/Zulu', '/home/conda/feedstock_root/build_artifacts/xgboost-split_1690228415297/_build_env/x86_64-conda-linux-gnu/sysroot/usr/share/zoneinfo/right/WET']'
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/R/lib/libR.so found in conda-forge::r-base-4.2.3-hb1b9b86_6
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-13.1.0-hfd8a6a1_0
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libgomp.so.1 found in conda-forge::_openmp_mutex-4.5-2_gnu
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-13.1.0-he5830b7_0
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
   INFO (r-xgboost,lib/R/library/xgboost/libs/xgboost.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/ld-linux-x86-64.so.2 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_16
WARNING (r-xgboost): plugin library (R) package conda-forge::r-jsonlite-1.8.7-r42h57805ef_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-magrittr-2.0.3-r42h57805ef_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-matrix-1.6_0-r42h316c678_0 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
WARNING (r-xgboost): plugin library (R) package conda-forge::r-data.table-1.14.8-r42h029312a_2 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

The warning highlighted in the previous comment is from the `libxgboost` output section. That is, it is an existing problem in the recipe, in a section that builds *prior to* - i.e., *independent of* - any of the PR changes.

My experience with R recipes is that, despite what the documentation may say, OpenMP should be declared in `host` and only there. Declaring it in other locations (e.g., `build` + `run`) leads to problems, specifically, final metadata (e.g., `llvm-openmp>=16`) being incompatible with globally pinned version (e.g., `llvm-openmp=15`), leading to builds that are unusable in downstream solves. Perhaps this doesn't translate to library builds (`libxgboost`), but it is my working practice for R packages.

If this is misguided, please elaborate (the current documentation on OpenMP linking is merely prescriptive without any explanation).

File: recipe/meta.yaml (hunk 3 above)
Comment (jakirkham) [reply recipe/meta.yaml:159]: Should add [the conda-forge docs]( https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp ) also suggest OpenMP should be added to `build`

File: recipe/meta.yaml (hunk 4 above)
Comment (jakirkham) [reply recipe/meta.yaml:186]: Ok agree that `llvm-openmp` & `libgomp` should work via `run_exports` and shouldn't be needed in `run`

Also agree that `r-xgboost` looks ok and this is a `libxgboost` issue

File: recipe/meta.yaml (hunk 3 above)
Comment (jakirkham) [reply recipe/meta.yaml:159]: Yeah the fact that `run_exports/strong` is used by [`llvm-opemp`]( https://github.com/conda-forge/openmp-feedstock/blob/2e3d1be71c92b2a586ef4c1e3100b1c23587558d/recipe/meta.yaml#L30-L32 ) & [`libgomp`]( https://github.com/conda-forge/ctng-compilers-feedstock/blob/923b25fa711bd71cc7c43299e032bd87579dbce9/recipe/meta.yaml#L440-L442 ) suggests the intended usage is to add both of these to `build` and nowhere else as `run_exports` will add them to `host` & `run` automatically

Could we please make that change here?

Maybe a separate `r-base` issue should be raised to discuss some of the pain points encountered with OpenMP & R to-date

Comment (mfansler) [reply recipe/meta.yaml:159]: Thanks for this clarification! I wasn't previously aware of the strong-weak exporting distinction. 

Perhaps [the problem I was seeing with the R recipes](https://matrix.to/#/!SOyumkgPRWoXfQYIFH:matrix.org/$168693130842PDzPS:gitter.im?via=matrix.org&via=gitter.im&via=cadair.com) was the fact that `llvm-openmp` was included in the `run:` section of the recipe explicitly and that was leading to an overly strict constraint.

I will revert it back here.

Comment (jakirkham) [reply recipe/meta.yaml:159]: Ah ok. Yeah adding to `run` seems odd. So it makes sense that it may cause issues

Thanks! 🙏

**Comment by jakirkham** [comment]
Thanks Mervin and everyone for reviewing! 🙏

Let's get this out

We can follow up on other items (like cleaning up OpenMP dependencies outside `r-xgboost`) separately. Already have some changes related to issues like these that we can discuss after

**Pull request merged**